mailpoet/lib/Cron/CronWorkerRunner.php

```diff
@@ -72,6 +73,8 @@ public function run(CronWorkerInterface $worker) {
       }
     } catch (\Exception $e) {
       if (isset($task) && $task && $e->getCode() !== CronHelper::DAEMON_EXECUTION_LIMIT_REACHED) {
+
+        LoggerFactory::getInstance()->getLogger()->error($e->getMessage(), ['error' => $e]);
```

Comment:
I guess both of these loggers should define a topic and a message.

I've just seen a similar thing in a PR that introduced a `cron` loggin topic: https://github.com/mailpoet/mailpoet/pull/4403/commits/c30f6a385f264670d73cef67973a44bd3a82d80d

Unfortunately not merged yet.

Additionally, can we use DI for the logger factory in all the new cases? See the link above, how it's used there.

Reply:
> should define a topic
For the topic, I will add an Todo for now. Once its merged, we use cron.

> and a message

Isn't the message `$e->getMessage()`?

> can we use DI

Thanks for catching, will do.

Reply:
> For the topic, I will add an Todo for now. Once its merged, we use cron.

I think you can add the constant as well. If you do it exactly on the same line as [here](https://github.com/mailpoet/mailpoet/pull/4403/commits/c30f6a385f264670d73cef67973a44bd3a82d80d#diff-eba49c637a6dbaf40e3ce6928b8c026f3e14aae04a8abf559beec797c0d09ef2R34), then I think the rebase of the other branch will even pass without conflicts — seeing both branches have the same change. Do you want to add it or prefer to do it later (an extra PR)?

> Isn't the message $e->getMessage()?

Oh, I missed that. I thought that we always pass in our own text, but now I see we seem to do that only for `info`, for error logs this is fine.

Reply:
later in an extra PR